src/components/VideoEditor.tsx

```diff
@@ -0,0 +1,229 @@
+import React, { useState } from 'react';
+import '../index.css';
+import { Ambiance } from '../config/ambiance/types';
+import Button from '@mui/material/Button';
+import TextField from '@mui/material/TextField';
+import MenuItem from '@mui/material/MenuItem';
+import Stack from '@mui/material/Stack';
+import Alert from '@mui/material/Alert';
+import AlertTitle from '@mui/material/AlertTitle';
+import { Channel } from '../config/ambiance/channels';
+
+
+type VideoEditorProps = {
+  documentId: string,
+  video?: Partial<Ambiance>,
+  videoUrlCode?: string,
+  editVideo?: (documentId: string, videoUrlCode: string, newData: Ambiance) => void,
+  deleteVideo?: (documentId: string, videoUrlCode: string) => void,
+  addVideo?: (documentId: string, newData: Ambiance) => void
+}
+
+
+export default function VideoEditor(props: VideoEditorProps) {
+  const { video, documentId, videoUrlCode, editVideo, deleteVideo, addVideo } = props;
+  const [showDeleteAlert, setShowDeleteAlert] = useState<Boolean>(false);
+  const [showEditAlert, setShowEditAlert] = useState<Boolean>(false);
+  const [vidData, setVidData] = useState<Partial<Ambiance> | undefined>(video);
+
+  const verifyDelete = () => {
+    setShowDeleteAlert(true);
+  }
+
+  const verifyEdit = () => {
+    setShowEditAlert(true);
+  }
+
+  const deleteAction = () => {
+    if (!videoUrlCode || !deleteVideo) {
+      return;
+    }
+
+    deleteVideo(documentId, videoUrlCode);
+    setShowDeleteAlert(false);
+  }
+
+  const editVideoAction = () => {
+    if (!vidData || !videoUrlCode || !editVideo) {
+      return;
+    }
+
+    if (!vidData.name || !vidData.code || !vidData.group) {
+      alert("Name, Code, and Group are required");
+      return;
+    }
```

Comment:
Would be nice to actually have this component be setup with a Form with required fields which wouldnt let you submit if these fields were not set. Can we create a ticket for that and work on it before moving onto the top level category add/edit/delete?

Reply:
These checks would probably still be necessary out of safety, but the Form would provide a much better user experience

Reply:
Also as part of that ticket can we clean up the input UI? Like labels on top, required fields marked, cleaner MUI input styles (non-filled)
![Screenshot 2024-04-05 at 10 42 46 AM](https://github.com/hermanator608/ambiance/assets/29784945/d1c51384-3a98-439e-a7d9-cc06b621c156)